File: TaboolaSDKExampleObjective-C/TaboolaSDKExampleObjective-C/SDK Web/WebViewController.m
//
//  WebViewController.m
//  TaboolaSDKExampleObjective-C
//
//  Copyright © 2020 Taboola. All rights reserved.
//

#import "WebViewController.h"
#import <TaboolaSDK/TaboolaSDK.h>
#import <WebKit/WebKit.h>


@interface WebViewController () <WKNavigationDelegate, TBLWebPageDelegate>

@property (weak, nonatomic) IBOutlet UIView *webViewContainer;

@property (nonatomic, strong) WKWebView *webView;

// The TBLWebPage object that will contain the Taboola content fetched via JS
@property (nonatomic, strong) TBLWebPage* webPage;
@property (nonatomic, strong) TBLWebUnit *webUnit;

@end

@implementation WebViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _webView = [[WKWebView alloc] initWithFrame:CGRectZero];
    
    _webView.frame = self.view.frame;
    [_webViewContainer addSubview:_webView];
    
    _webPage = [[TBLWebPage alloc]initWithDelegate:self];
    _webUnit = [_webPage createUnitWithWebView:_webView];
    
    [self loadExamplePage:_webView];
}

- (void)loadExamplePage:(WKWebView*)webView {
    NSString* htmlPath = [[NSBundle mainBundle] pathForResource:@"SampleHTMLPage" ofType:@"html"];
    NSString* appHtml = [NSString stringWithContentsOfFile:htmlPath encoding:NSUTF8StringEncoding error:nil];
    [webView loadHTMLString:appHtml baseURL:[NSURL URLWithString:@"https://cdn.taboola.com/mobile-sdk/init/"]];	
}

- (void)webView:(UIView *)webView didLoadPlacementNamed:(NSString *)placementName withHeight:(CGFloat)height {
    NSLog(@"%@", placementName);
}

- (void)taboolaView:(UIView *)taboolaView didFailToLoadPlacementNamed:(NSString *)placementName withErrorMessage:(NSString *)error {
    NSLog(@"%@", error);
}

- (BOOL)webView:(WKWebView *)webView didClickPlacementName:(NSString *)placementName
         itemId:(NSString *)itemId
       clickUrl:(NSString *)clickUrl
      isOrganic:(BOOL)organic
     customData:(NSDictionary *)customData {
    // Return 'YES' for Taboola SDK to handle the click event (default behavior).
    // Return 'NO' to handle the click event yourself. (Applicable for organic content only (including Audience Exchange items).)
    
    // You can identify Audience Exchange items by checking the customData dictionary
    // Note that you will also need to pass in an Extra Property flag called allowAudienceExchangeClickOverride with the value of true
    if (customData[@"isAudienceExchange"] && [customData[@"isAudienceExchange"] boolValue]) {
        // Handle Audience Exchange items
        return NO;
    }
    
    return YES;
}

@end

File: TaboolaSDKExampleSwift/TaboolaSDKExampleSwift/SDK Web/WebViewController.swift
//
//  WebViewController.swift
//  TaboolaSDKExampleV4
//
//  Copyright © 2020 Taboola. All rights reserved.
//

import Foundation
import TaboolaSDK

class WebViewController: UIViewController, WKNavigationDelegate {
    @IBOutlet weak var webViewContainer: UIView!
    var webView = WKWebView()
    
    // The TBLWebPage object that will contain the Taboola content fetched via JS
    var webPage: TBLWebPage?
    var webUnit: TBLWebUnit?

    override func viewDidLoad() {
        super.viewDidLoad()
        
        webView.frame = view.frame
        webViewContainer.addSubview(webView)
        
        webPage =  TBLWebPage.init(delegate: self)
        webUnit = webPage?.createUnit(with: webView)
                
        try? loadExamplePage()
    }
    
    func loadExamplePage() throws {
        guard let htmlPath = Bundle.main.path(forResource: "SampleHTMLPage", ofType: "html") else {
            print("Error loading HTML")
            return
        }
        let appHtml = try String.init(contentsOfFile: htmlPath, encoding: .utf8)
        webView.loadHTMLString(appHtml, baseURL: URL(string: "https://cdn.taboola.com/mobile-sdk/init/"))
    }
}

extension WebViewController: TBLWebPageDelegate {
    func webView(_ webView: WKWebView!, didLoadPlacementName placementName: String!, height: CGFloat) {
        print("Placement name: \(String(describing: placementName)) has been loaded with height: \(height)")
    }
    
    func webView(_ webView: WKWebView!, didFailToLoadPlacementName placementName: String!, errorMessage error: String!) {
        print("Placement name: \(String(describing: placementName)) failed to load!)")
        print("Error: \(String(describing: error))")
    }
    
    func webView(_ webView: WKWebView!, didClickPlacementName placementName: String!, itemId: String!, clickUrl: String!, isOrganic organic: Bool) -> Bool {
        // Return 'true' for Taboola SDK to handle the click event (default behavior).
        // Return 'false' to handle the click event yourself. (Applicable for organic content only.)
        return true
    }
    
    func webView(_ webView: WKWebView!, didClickPlacementName placementName: String!, itemId: String!, clickUrl: String!, isOrganic organic: Bool, customData: [AnyHashable : Any]!) -> Bool{
        // Return 'true' for Taboola SDK to handle the click event (default behavior).
        // Return 'NO' to handle the click event yourself. (Applicable for organic content only (including Audience Exchange items).)
        
        // You can identify Audience Exchange items by checking the customData dictionary
        // Note that you will also need to pass in an Extra Property flag called allowAudienceExchangeClickOverride with the value of true
        if let isAudienceExchange = customData["isAudienceExchange"] as? Bool {
            if isAudienceExchange == true {
                // Handle Audience Exchange items
                return false
            }
        }

        
        return true
    }
}
